src/components/SplitPane.jsx

```diff
@@ -0,0 +1,146 @@
+var React = require('react'),
+    classNames = require('classnames'),
+    Pane = require('./Pane.jsx'),
+    Resizer = require('./Resizer.jsx'),
+    style = require('./component-playground.less');
+
+export default React.createClass({
+  displayName: 'SplitPane',
+
+  propTypes: {
+    minSize: React.PropTypes.number,
+    defaultSize: React.PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.number,
+    ]),
+    split: React.PropTypes.string,
+    onChange: React.PropTypes.func
+  },
+
+  getInitialState: function() {
+    return {
+      active: false,
+      resized: false
+    };
+  },
+
+  getDefaultProps: function() {
+    return {
+      minSize: 0
+    };
+  },
+
+  componentDidMount: function() {
+    document.addEventListener('mouseup', this.onMouseUp);
+    document.addEventListener('mousemove', this.onMouseMove);
+    var ref = this.refs.pane1;
+    if (ref && this.props.defaultSize && !this.state.resized) {
+      this.setState({
+        size: this.props.defaultSize
+      });
+    }
+  },
+
+  componentWillUnmount: function() {
+    document.removeEventListener('mouseup', this.onMouseUp);
+    document.removeEventListener('mousemove', this.onMouseMove);
+  },
+
+  onMouseDown: function() {
+    var position = this.props.split === 'vertical' ?
+     event.clientX : event.clientY;
+    this.setState({
+      active: true,
+      position: position
+    });
+  },
+
+  onMouseMove: function() {
+    if (this.state.active) {
+      var ref = this.refs.pane1;
+      if (ref) {
+        var node = ref.getDOMNode();
+        var width = node.offsetWidth;
+        var height = node.offsetHeight;
+        var current = this.props.split === 'vertical' ? event.clientX
+                                                      : event.clientY;
+        var size = this.props.split === 'vertical' ? width : height;
```

Comment:
I think we can use `node.offsetWidth` directly because the variable is used only once.